faq: add an entry for BIOS/UEFI boot modes (#102)
coreos/fedora-coreos-docs: modules/ROOT/pages/faq.adoc

    - name: docker.service
      mask: true
----

> == Are Fedora CoreOS disk images hybrid BIOS+UEFI bootable?

Reviewer: ```suggestion
== Are Fedora CoreOS x86_64 disk images hybrid BIOS+UEFI bootable?
```

> The bare metal images we provide can be used for either BIOS (legacy) boot or UEFI boot,

Reviewer: Hey @nasirhm. I did a little more digging on this topic and updated my forum reponse: https://discussion.fedoraproject.org/t/are-fedora-coreos-disk-images-hybrid-bios-uefi-bootable/21911/2?u=dustymabe

```suggestion
The x86_64 images we provide can be used for either BIOS (legacy) boot or UEFI boot. They contain a hybrid BIOS/UEFI partition setup that allows them to be used for either. The exception to that is the `metal4k` 4k native image, which is targeted at disks with 4k sectors and [does not have a BIOS boot partition](https://github.com/coreos/coreos-assembler/blob/12029fea7798fa5d3535eafcf8c3d02f9a6095e4/src/cmd-buildextend-metal#L200-L202) because 4k native disks are [only supported with UEFI](https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/hard-drives-and-partitions#advanced-format-drives).
```

> The bare metal images we provide can be used for either BIOS (legacy) boot or UEFI boot, You can boot it in either BIOS (legacy) or UEFI mode regardless of what mode the OS will use once installed.

Reviewer 2: I think you mixed up a bunch of statements here, the second part is usually found in the context of booting the live-ISO (in order to image a bare-metal disk).

> The x86_64 images we provide can be used for either BIOS (legacy) boot or UEFI boot. They contain a hybrid BIOS/UEFI partition setup that allows them to be used for either. The exception to that is the `metal4k` 4k native image, which is targeted at disks with 4k sectors and https://github.com/coreos/coreos-assembler/blob/12029fea7798fa5d3535eafcf8c3d02f9a6095e4/src/cmd-buildextend-metal#L200-L202[does not have a BIOS boot partition] because 4k native disks are https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/hard-drives-and-partitions#advanced-format-drives[only supported with UEFI].

Reviewer 3: Not sure if it's worth linking to cosa code, but meh. Hmm, also wish we had a better link than the Microsoft one, but I can't find one offhand, so WFM! (I found https://lwn.net/Articles/377895/ though that might be too detailed.)

Reviewer: Yeah the existing link is fine. I'm sure it'll break at some point and we'll fix it.

> The x86_64 images we provide can be used for either BIOS (legacy) boot or UEFI boot. They contain a hybrid BIOS/UEFI partition setup that allows them to be used for either. The exception to that is the `metal4k` 4k native image, which is targeted at disks with 4k sectors and https://github.com/coreos/coreos-assembler/blob/12029fea7798fa5d3535eafcf8c3d02f9a6095e4/src/cmd-buildextend-metal#L200-L202[does not have a BIOS boot partition] because 4k native disks are https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/hard-drives-and-partitions#advanced-format-drives[only supported with UEFI].

Reviewer 3: Maybe for the first sentence we can add the words "bare metal" since that's how it's labeled in the download page, e.g.:

```
The x86_64 bare metal images we provide can be used for either BIOS (legacy) boot or UEFI boot.
```

?

Reviewer: I think we should keep it generic. I would have claimed otherwise, but the truth is that this isn't limited to our bare metal images. From looking at the code the only x86_64 image we create that isn't hybrid is the 4kn disk image. Am I reading it correctly?

Reviewer 3: Sure, that's true. Though OTOH it's not like it matters much for most other platforms, and although I love the consistency, it's possible we may change it in the future (see e.g. https://github.com/coreos/fedora-coreos-config/pull/407) and should reverse the right to break that.

No strong opinion either way though.

Reviewer: Thanks. Let's merge as-is then.

> The x86_64 images we provide can be used for either BIOS (legacy) boot or UEFI boot. They contain a hybrid BIOS/UEFI partition setup that allows them to be used for either. The exception to that is the `metal4k` 4k native image, which is targeted at disks with 4k sectors and [does not have a BIOS boot partition](https://github.com/coreos/coreos-assembler/blob/12029fea7798fa5d3535eafcf8c3d02f9a6095e4/src/cmd-buildextend-metal#L200-L202) because 4k native disks are [only supported with UEFI](https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/hard-drives-and-partitions#advanced-format-drives).

Reviewer: ```suggestion
The x86_64 images we provide can be used for either BIOS (legacy) boot or UEFI boot. They contain a hybrid BIOS/UEFI partition setup that allows them to be used for either. The exception to that is the `metal4k` 4k native image, which is targeted at disks with 4k sectors and https://github.com/coreos/coreos-assembler/blob/12029fea7798fa5d3535eafcf8c3d02f9a6095e4/src/cmd-buildextend-metal#L200-L202[does not have a BIOS boot partition] because 4k native disks are https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/hard-drives-and-partitions#advanced-format-drives[only supported with UEFI].
```
